### DISCUSSION TITLE
Connecting to heroku postgresql without database url

### DISCUSSION THREAD

**Discussion by rjtngit** [prisma/prisma#8086] [Q&A] [answered]
Hey, I'm setting up a full stack website using vercel, nextjs, postgresql, and prisma. I followed this guide: https://vercel.com/guides/nextjs-prisma-postgres

The heroku database url changes occasionally and is not recommended to be used. Does anyone know how I can connect prisma to a heroku database without hard coding the database url as an environment variable?

**Comment by ryands17** [comment, marked as answer]
@rempelj 👋 

I don't know much about Heroku's database, but it would be difficult working with an always changing URL. In any case, you can set the URL dynamically in [this](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#datasources) way, but you would need to check if the URL is still valid or not and if not, assign the new URL again.

I would still suggest using another database whose URL doesn't change.

**Reply by rjtngit** [reply]
Thanks @ryands17. Do you have any suggestions? I am using a free hobby account with Heroku.

**Reply by ryands17** [reply]
Just possibly adding the URL above or switching to AWS RDS if you're in the free tier. @janpio wdyt?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
